qiskit/synthesis/__init__.py

```diff
@@ -78,6 +78,13 @@
 
    SolovayKitaevDecomposition
 
+SU(4) Synthesis
```

Comment:
Does anyone know what SU(4) is? Is it better to say something like "Two-qubit Unitary Synthesis"?

Reply:
```suggestion
SU(4) (2-qubit unitary) Synthesis
```

Reply:
I was sketching something out, please let me know what you think. Some of these are placeholders for things i'd like to add in the near future. 

```
qiskit.synthesis
 |--- su2
    |--- euler
    |--- nonorthogonal
 |--- su4
    |--- supercontrolled
    |--- controlled
    |--- siswap
    |--- b
 |--- unitary
    |--- aqc (approximate quantum compiling)
    |--- qsd (quantum shannon decomposition)
    |--- ccd (column-by-column)
 |--- state
    |
 |--- clifford
    |--- ag
    |--- bm
    |--- full
    |--- greedy
    |--- layers
 |--- linear
    |--- graysynth
    |--- depth_lnn
    |--- pmh
 |--- permutation  * use method/paper names here
    |--- lnn
    |--- full
 |--- cnotdihedral
    |--- full
    |--- general
    |--- two_qubits
 |--- discrete_basis
    |--- solovay-kitaev
 |--- evolution
    |--- lie_trotter
    |--- suzuki-trotter
    |--- product_formula
    |--- qdrift
 |--- povm
 ```

Reply:
Nice! I'll open an issue to discuss the structure of `qiskit/synthesis` (as this is not a part of this PR).

Reply:
See: https://github.com/Qiskit/qiskit-terra/issues/9667